src/evict/evict_page.c

```diff
@@ -663,34 +663,64 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     /* Make sure that both conditions above are not true at the same time. */
     WT_ASSERT(session, !use_snapshot_for_app_thread || !is_eviction_thread);
 
-    if (!conn->txn_global.checkpoint_running && !WT_IS_HS(btree->dhandle) &&
-      (use_snapshot_for_app_thread || is_eviction_thread)) {
+    /* History store data are always evictable. */
+    if (WT_IS_HS(btree->dhandle)) {
+        /*
+         * FIX-ME-WT-5316: remove this when we have removed history store score and
+         * __rec_update_stable. No need to set visibility flag for history store.
+         */
+        if (!WT_SESSION_BTREE_SYNC(session))
+            LF_SET(WT_REC_VISIBLE_ALL);
+    } else {
         if (is_eviction_thread) {
-            /*
-             * Eviction threads do not need to pin anything in the cache. We have a exclusive lock
-             * for the page being evicted so we are sure that the page will always be there while it
-             * is being processed. Therefore, we use snapshot API that doesn't publish shared IDs to
-             * the outside world.
-             */
-            __wt_txn_bump_snapshot(session);
-            snapshot_acquired = true;
-
-            /*
-             * Make sure once more that there is no checkpoint running. A new checkpoint might have
-             * started between previous check and acquiring snapshot. If there is a checkpoint
-             * running, release the snapshot and fallback to global visibility checks.
-             */
-            if (conn->txn_global.checkpoint_running) {
-                __wt_txn_release_snapshot(session);
-                snapshot_acquired = false;
+            /* Eviction thread doing eviction. */
+            if (!conn->txn_global.checkpoint_running) {
+                /*
+                 * Eviction threads do not need to pin anything in the cache. We have a exclusive
```

Comment:
I know this wasn't your change but can we fix this one too? "We have an exclusive lock", not "We have a exclusive lock"

Reply:
done